src/DotLiquid/StandardFilters.cs

```diff
@@ -931,6 +931,43 @@ public static IEnumerable Compact(object input)
             ary.RemoveAll(item => item == null);
             return ary;
         }
+
+        /// <summary>
+        /// Creates an array including only the objects with a given property value, or any truthy value by default.
+        /// </summary>
+        /// <param name="input">an array to be filtered</param>
+        /// <param name="propertyName">The name of the property to filter by</param>
+        /// <param name="target_value">Value to retain, if null object containing this property are retained</param>
+        /// <returns></returns>
+        public static IEnumerable Where(IEnumerable input, string propertyName, string target_value = null)
+        {
+            if (input == null)
+                return null;
+
+            if (propertyName.IsNullOrWhiteSpace())
+                throw new ArgumentNullException(paramName: nameof(propertyName), message: $"'{nameof(propertyName)}' cannot be null or empty.");
+
+            return input.Cast<object>().Where(source => source.HasMatchingProperty(propertyName, target_value));
+        }
+
+        /// <summary>
+        /// Checks if the given object has a matching property name.
+        /// * If target_value is provided, then the property is compared to target_value
+        /// * If target_value is null, then the property is checked for "Truthyness".
+        /// <param name="any">an object to be assessed</param>
+        /// <param name="propertyName">The name of the property to test for</param>
+        /// <param name="target_value">target property value</param>
+        /// <returns></returns>
+        /// </summary>
+        private static bool HasMatchingProperty(this object any, string propertyName, string target_value)
+        {
+            object propertyValue = any is IDictionary dictionary ? dictionary[propertyName]

```

Comment:
The dictionary array accessor will throw KeyNotFoundException if the property is not found. You probably didn't plan for exceptions here.

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.item?view=net-5.0#exceptions

 You probably want to use ContainsKey instead 

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.containskey?view=net-5.0

Reply:
And if this is comment is accurate, there is likely a gap in the test cases.

Reply:
Reinstated original if/else block with contains(key).
I can't spot the gap in test cases, could you elaborate on the case I've missed?

Reply:
The missing case is any is implementing IDictionary but dictionary does not contain propertyName. Should result in dictionary[propertyName] throws KeyNotFoundException

Reply:
That test case is already present see "Tartan flat cap" in TestWhere_ShopifySample3.